Refit/RequestBuilderImplementation.cs

```diff
@@ -32,23 +33,80 @@ public RequestBuilderImplementation(Type targetInterface, RefitSettings refitSet
             }
 
             targetType = targetInterface;
-            interfaceHttpMethods = targetInterface.GetMethods()
-                                                  .Where(x => x.GetCustomAttributes(true)
-                                                               .OfType<HttpMethodAttribute>().Any())
-                                                  .Select(x => new RestMethodInfo(targetInterface, x, settings))
-                                                  .ToDictionary(k => k.Name, v => v);
+
+            foreach (var methodInfo in targetInterface.GetMethods()) {
+                var attrs = methodInfo.GetCustomAttributes(true);
+                var hasHttpMethod = attrs.OfType<HttpMethodAttribute>().Any();
+                if (hasHttpMethod) {
+                    var restinfo = new RestMethodInfo(targetInterface, methodInfo, settings);
+                    interfaceHttpMethods.AddOrUpdate(methodInfo.Name, s => new[] { restinfo }.ToList(), (s, list) => {
+                        list.Add(restinfo);
+                        return list;
+                    });
+                }
+            }
+
         }
 
         public IEnumerable<string> InterfaceHttpMethods => interfaceHttpMethods.Keys;
 
-        public Func<HttpClient, object[], object> BuildRestResultFuncForMethod(string methodName)
+        public Func<HttpClient, object[], object> GetHttpMethod(string key, object[] parameters = null)
+        {
+            var parameterTypes = parameters?.Select(p => p?.GetType() ?? typeof(object)).ToArray();
```

Comment:
is the `p?` necessary inside the select? Should it be an error if they put a `null` in the `Type` array?

Reply:
I dont't think it should be an error.
The parameterTypes variable is generated from "objects" which are set as parameters to the RestService and  i think its totally possible to set a Parameter to null...

Reply:
This can't be based on the runtime types, it'd need to be based on the types of the method parameters as defined?

Reply:
Ok, i understand!
Then i think we get get rid of "GetHttpMethod"...
The only thing this method has done was to build the parameterTypes Array.
I'll change it in the code, please check if thats a better approach!